src/Shared/Model/NuGetMetadata.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.Model;
+
+using Newtonsoft.Json;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading.Tasks;
+
+/// <summary>
+/// A class to represent Package Metadata for a NuGet package.
+/// </summary>
+public class NuGetMetadata : IPackageSearchMetadata
```

Comment:
Nit: [immutable record types](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record) offer some nice properties - do you think this could be made an immutable record class?

Reply:
Resolved in new PR.